chatkit-android/src/main/kotlin/com/pusher/chatkit/BeamsPushNotifications.kt

```diff
@@ -0,0 +1,77 @@
+package com.pusher.chatkit
+
+import android.content.Context
+import com.pusher.chatkit.pushnotifications.BeamsTokenProviderService
+import com.pusher.chatkit.pushnotifications.PushNotifications
+import com.pusher.pushnotifications.Callback
+import com.pusher.pushnotifications.PusherCallbackError
+import com.pusher.pushnotifications.auth.TokenProvider
+import com.pusher.util.Result
+import elements.Error
+import java8.util.concurrent.CompletableFuture
```

Comment:
Is this some kind of shim for pre-java8 support?

The rest of the library uses Pablo's implementation which should be  sufficient without additional dependencies (and sharing the existing backing threadpool)

Reply:
Yes, Android 24 has this (https://developer.android.com/reference/java/util/concurrent/CompletableFuture) but that's really recent in Android years.

This is a good way to make async calls synchronous, something not possible with Pablo's abstractions.